### PR TITLE
Clarify Password Change Failure Message

### LINKED ISSUE #2412: Clarify Password Change Failure Message
**What type(s) of user does this feature affect?**
 - volunteers
 - supervisors
 - admins

**Description**
Currently when a user's password change fails due to the new password not complying with the minimum length requirement a message is shown that states "1 error prohibited this (Volunteer/Casa admin/Supervisor) from being saved: " and then the reason listed below in a bullet point. 

Regardless of who is trying to change their password, if the password change fails the text should read "1 (or whatever the appropriate number of errors is) error(s) prohibited this password change from being saved: " and then the reason or reasons listed below in a bulleted list. 

**Screenshots of current behavior, if any**  
Error message unclear

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 123456

### PR DESCRIPTION
**Pull request by AravindSelvamani**
### What GitHub issue is this PR for if any?
Resolves #2412

### What changed, and why?
Modified the error message for the password update event. Regardless of who is trying to change their password, if the password change fails the text should read "1 (or whatever the appropriate number of errors is) error(s) prohibited this password change from being saved: " and then the reason or reasons listed below in a bulleted list.

Currently, when a user's password change fails due to the new password not complying with the minimum length requirement a message is shown that states "1 error prohibited this (Volunteer/Casa admin/Supervisor) from being saved: " and then the reason listed below in a bullet point.

### How will this affect user permissions?
- Volunteer 
- Supervisor
- Admin

### Screenshots please :)
Screenshot 2021-10-01 at 9 33 18 PM

### CODE REVIEW

File: app/views/shared/_error_messages.html.erb [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-      saved:</h6>
-
-    <ul>
-      <% resource.errors.full_messages.each do |message| %>
-        <li><%= message %></li>
-      <% end %>
-    </ul>
-  </div>
+  <% full_error_messages = resource.errors.full_messages %>
+  <% if full_error_messages.join(' ').downcase.include? "password" %>
```
Comment (compwron) [inline app/views/shared/_error_messages.html.erb:3]: Please refactor this to keep logic in the controller (or a controller helper) and out of the erb files where possible, or open a github issue to clean up after this PR so we remember to do it :)

File: spec/system/users/edit_spec.rb [hunk 2]
```diff
@@ -141,7 +141,7 @@
       fill_in "Password Confirmation", with: "1234"
 
       click_on "Update Password"
-
+      expect(page).to have_content "2 errors prohibited this password change from being saved:"
```
Comment (compwron) [inline spec/system/users/edit_spec.rb:144]: Yay tests :)

**Review by compwron — APPROVED** [review]

File: app/views/shared/_error_messages.html.erb (hunk 1 above)
Comment (AravindSelvamani) [reply app/views/shared/_error_messages.html.erb]: Moved the logic to the controller :)

Comment (compwron) [reply app/views/shared/_error_messages.html.erb]: :)

**Pull request merged**